framework/interface/urls.py

```diff
@@ -34,6 +34,7 @@ def get_handlers():
         tornado.web.url(r'/api/targets/([0-9]+)/transactions/hrt/?([0-9]+)?/?$', api_handlers.TransactionHrtHandler, name='transactions_hrt_api_url'),
         tornado.web.url(r'/api/targets/([0-9]+)/poutput/?' + plugin_group_re + '/?' + plugin_type_re + '/?' + plugin_code_re + '/?$', api_handlers.PluginOutputHandler, name='poutput_api_url'),
         tornado.web.url(r'/api/targets/([0-9]+)/poutput/names/?' + plugin_group_re + '/?' + plugin_type_re + '/?' + plugin_code_re + '/?$', api_handlers.PluginNameOutput, name='plugin_name_api_url'),
+        tornado.web.url(r'/api/targets/([0-9]+)/export/?$', api_handlers.ReportExportHandler, name='report_export_api_url'),
```

Comment:
@DePierre Actually, we have plans to separate the React UI from Tornado API server. Means Render everything using React rather than Jinja2. We will make backend as simple API server. 

Now, as you suggested API should be something like this -> `/api/targets/{id}/export/{template}`  , but backend server will not have any information about templates, as the export functionality is completely client side and only React or UI will have information about available templates. 
 
Use of `/api/targets/{id}/export/` is to just return data from DB, `docxtemplater` will do the rest. I mean we don't need to have seperate api for template, because the same API will work for all template.